Docs/developer_guide/parameter_nodes.md

```diff
@@ -110,20 +112,24 @@ This will make the default value of the `numIterations` parameter 500. If the `n
-| `pathlib.WindowsPath`                            | `pathlib.WindowsPath()` (which is the current directory)     |
-| `pathlib.PurePath`                               | `pathlib.PurePath()` (which is the current directory)        |
-| `pathlib.PurePosixPath`                          | `pathlib.PurePosixPath()` (which is the current directory)   |
-| `pathlib.PureWindowsPath`                        | `pathlib.PureWindowsPath()` (which is the current directory) |
+| Type                                             | Implicit default                                                                       |
+|--------------------------------------------------|----------------------------------------------------------------------------------------|
+| `int`                                            | `0`                                                                                    |
+| `float`                                          | `0.0`                                                                                  |
+| `str`                                            | `""`                                                                                   |
+| `bool`                                           | `False`                                                                                |
+| `vtkMRMLNode` (including subclasses)             | `None`                                                                                 |
+| `list` (hinted as `list[int]`, `list[str]`, etc) | `[]` (empty list)                                                                      |
+| `tuple` (hinted as `tuple[int, bool]`, etc)      |  A tuple of the defaults of all the elements (e.g. `tuple[int, bool]` -> `(0, False)`) |
+| `dict` (hinted as `dict[keyType, valueType]`)    | `{}` (empty dictionary)                                                                |
+| `pathlib.Path`                                   | `pathlib.Path()` (which is the current directory)                                      |
+| `pathlib.PosixPath`                              | `pathlib.PosixPath()` (which is the current directory)                                 |
+| `pathlib.WindowsPath`                            | `pathlib.WindowsPath()` (which is the current directory)                               |
+| `pathlib.PurePath`                               | `pathlib.PurePath()` (which is the current directory)                                  |
+| `pathlib.PurePosixPath`                          | `pathlib.PurePosixPath()` (which is the current directory)                             |
+| `pathlib.PureWindowsPath`                        | `pathlib.PureWindowsPath()` (which is the current directory)                           |
+
+Note: For specifying the default of a tuple, use `Annotated[tuple[int, bool], Default((4, True))]`, not `tuple[Annotated[int, Default(4)], Annotated[bool, Default(True)]]`.
```

Comment:
Is it possible to support this? It seems like it would "just work" but if there's some technical reason why it doesn't maybe add a comment about why and a runtime warning suggesting the correct approach.

Reply:
There are a couple reasons I chose not to support this.

Interface consistency:

- Doing `tuple[Annotated[int, Default(4)], Annotated[bool, Default(True)]]` may make sense, but it is the only container that it currently makes sense for.
    - Doing `list[Annotated[int, Default(4)]]` is confusing. What is defaulted?
    - Similarly `dict[Annotated[str, Default("hi")], Annotated[int, Default(4)]]` doesn't make sense.
    - However, in each of these cases setting a default for the whole value makes sense as opposed to piecewise values (`Annotated[list[int], Default([4])]` or `Annotated[dict[str, int], Default({"hi": 4})` both make sense).

Technical reasons:

- I didn't want the serializers to have to deal with either names or custom default values (the `Default` annotation). That seemed like it should be something else's concern.
    - `_ParameterInfo`/`_Parameter`/`_CachedParameter` in `wrapper.py` were added to deal with these.
    - For simplicity, it makes sense for the composite serializers (list, tuple, dict) to just deal with serializers and not the `_Parameter` classes.
    - Work could be done to refactor somewhat and support this for tuple, but I think the added value is low since it does not add functionality, just multiple ways of doing the same thing.

Reply:
I can summarize this in the (docs? code?) for why this is not currently supported.